src/braket/ir/jaqcd/program_v1.py

```diff
@@ -59,40 +59,44 @@
 )
 from braket.schema_common import BraketSchemaBase, BraketSchemaHeader
 
-GateInstructions = Union[
-    CCNot,
-    CNot,
-    CPhaseShift,
-    CPhaseShift00,
-    CPhaseShift01,
-    CPhaseShift10,
-    CSwap,
-    CY,
-    CZ,
-    H,
-    I,
-    ISwap,
-    PhaseShift,
-    PSwap,
-    Rx,
-    Ry,
-    Rz,
-    S,
-    Swap,
-    Si,
-    T,
-    Ti,
-    Unitary,
-    V,
-    Vi,
-    X,
-    XX,
-    XY,
-    Y,
-    YY,
-    Z,
-    ZZ,
-]
+"""
+We need a constant lookup for our pydantic validator. Using a plain Union[] will result
+in an O(n) lookup for arbitrary payloads, having a negative impact on model creation times.
+"""
+_valid_gates = {
+    CCNot.Type.ccnot: CCNot,
+    CNot.Type.cnot: CNot,
+    CPhaseShift.Type.cphaseshift: CPhaseShift,
+    CPhaseShift00.Type.cphaseshift00: CPhaseShift00,
+    CPhaseShift01.Type.cphaseshift01: CPhaseShift01,
+    CPhaseShift10.Type.cphaseshift10: CPhaseShift10,
+    CSwap.Type.cswap: CSwap,
+    CY.Type.cy: CY,
+    CZ.Type.cz: CZ,
+    H.Type.h: H,
+    I.Type.i: I,
+    ISwap.Type.iswap: ISwap,
+    PhaseShift.Type.phaseshift: PhaseShift,
+    PSwap.Type.pswap: PSwap,
+    Rx.Type.rx: Rx,
+    Ry.Type.ry: Ry,
+    Rz.Type.rz: Rz,
+    S.Type.s: S,
+    Swap.Type.swap: Swap,
+    Si.Type.si: Si,
+    T.Type.t: T,
+    Ti.Type.ti: Ti,
+    Unitary.Type.unitary: Unitary,
+    V.Type.v: V,
+    Vi.Type.vi: Vi,
+    X.Type.x: X,
+    XX.Type.xx: XX,
+    XY.Type.xy: XY,
+    Y.Type.y: Y,
+    YY.Type.yy: YY,
+    Z.Type.z: Z,
+    ZZ.Type.zz: ZZ,
+}
 
 Results = Union[Amplitude, Expectation, Probability, Sample, StateVector, Variance]
```

Comment:
Won't see as much of a performance improvement, but it might be worth giving this the same treatment, as it might make a difference for very shallow circuits.